apollo-api/src/main/java/com/apollographql/apollo/api/Field.java

```diff
@@ -254,6 +382,9 @@ public ScalarType scalarType() {
     }
   }
 
+  /**
+   * Abstraction for a Field representing a conditional type.
+   */
   public static final class ConditionalTypeField extends Field {
```

Comment:
I feel docs can be certainly improved for ConditionalType. Can you suggest better docs here?